File: spring-boot-iot-ui/src/api/auditLog.ts
import { request } from './request'
import type { ApiEnvelope, IdType, PageResult, SystemErrorStats, BusinessAuditStats } from '../types/api'

export interface AuditLogRecord {
  id: IdType
  tenantId?: number
  userId?: number
  userName?: string
  traceId?: string
  deviceCode?: string
  productKey?: string
  operationModule?: string
  operationType?: string
  operationMethod?: string
  requestUrl?: string
  requestMethod?: string
  requestParams?: string
  responseResult?: string
  ipAddress?: string
  operationResult?: number
  resultMessage?: string
  errorCode?: string
  exceptionClass?: string
  operationTime?: string
  createTime?: string
}

export interface SystemErrorClusterRow {
  clusterKey: string
  operationModule?: string
  exceptionClass?: string
  errorCode?: string
  count: number
  distinctTraceCount: number
  distinctDeviceCount: number
  latestOperationTime?: string
  latestRequestUrl?: string
  latestRequestMethod?: string
  latestResultMessage?: string
}

export interface AuditLogQueryParams {
  userName?: string
  operationType?: string
  operationModule?: string
  requestMethod?: string
  requestUrl?: string
  traceId?: string
  deviceCode?: string
  productKey?: string
  resultMessage?: string
  errorCode?: string
  exceptionClass?: string
  operationResult?: number
  pageNum?: number
  pageSize?: number
  excludeSystemError?: boolean
}

function toQueryString(params: Record<string, unknown>) {
  const search = new URLSearchParams()
  Object.entries(params).forEach(([key, value]) => {
    if (value !== undefined && value !== null && value !== '') {
      search.append(key, String(value))
    }
  })
  return search.toString()
}

export function listLogs(params: AuditLogQueryParams = {}): Promise<ApiEnvelope<AuditLogRecord[]>> {
  const query = toQueryString(params)
  const path = `/api/system/audit-log/list${query ? `?${query}` : ''}`
  return request<AuditLogRecord[]>(path, { method: 'GET' })
}

export function pageLogs(params: AuditLogQueryParams = {}): Promise<ApiEnvelope<PageResult<AuditLogRecord>>> {
  const query = toQueryString(params)
  const path = `/api/system/audit-log/page${query ? `?${query}` : ''}`
  return request<PageResult<AuditLogRecord>>(path, { method: 'GET' })
}

export function getSystemErrorStats(params: AuditLogQueryParams = {}): Promise<ApiEnvelope<SystemErrorStats>> {
  const query = toQueryString(params)
  const path = `/api/system/audit-log/system-error/stats${query ? `?${query}` : ''}`
  return request<SystemErrorStats>(path, { method: 'GET' })
}

export function pageSystemErrorClusters(
  params: AuditLogQueryParams = {}
): Promise<ApiEnvelope<PageResult<SystemErrorClusterRow>>> {
  const query = toQueryString(params)
  const path = `/api/system/audit-log/system-error/clusters/page${query ? `?${query}` : ''}`
  return request<PageResult<SystemErrorClusterRow>>(path, { method: 'GET' })
}

export function getBusinessAuditStats(params: AuditLogQueryParams = {}): Promise<ApiEnvelope<BusinessAuditStats>> {
  const query = toQueryString(params)
  const path = `/api/system/audit-log/business/stats${query ? `?${query}` : ''}`
  return request<BusinessAuditStats>(path, { method: 'GET' })
}

export function getAuditLogById(id: IdType): Promise<ApiEnvelope<AuditLogRecord>> {
  return request<AuditLogRecord>(`/api/system/audit-log/get/${id}`, { method: 'GET' })
}

export function addAuditLog(data: Partial<AuditLogRecord>): Promise<ApiEnvelope<void>> {
  return request<void>('/api/system/audit-log/add', { method: 'POST', body: data })
}

export function deleteAuditLog(id: IdType): Promise<ApiEnvelope<void>> {
  return request<void>(`/api/system/audit-log/delete/${id}`, { method: 'DELETE' })
}

File: spring-boot-iot-system/src/main/java/com/ghlzm/iot/system/entity/AuditLog.java
package com.ghlzm.iot.system.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.io.Serializable;
import java.util.Date;

/**
 * 审计日志实体
 */
@Data
@TableName("sys_audit_log")
public class AuditLog implements Serializable {
      private static final long serialVersionUID = 1L;

      /**
       * 主键
       */
      @TableId(value = "id", type = IdType.ASSIGN_ID)
      private Long id;

      /**
       * 租户ID
       */
      private Long tenantId;

      /**
       * 操作用户ID
       */
      private Long userId;

      /**
       * 操作用户名称
       */
      private String userName;

      /**
       * 閾捐矾 traceId
       */
      private String traceId;

      /**
       * 鐩稿叧璁惧缂栫爜
       */
      private String deviceCode;

      /**
       * 鐩稿叧浜у搧 key
       */
      private String productKey;

      /**
       * 操作类型 insert/update/delete/select
       */
      private String operationType;

      /**
       * 操作模块
       */
      private String operationModule;

      /**
       * 操作方法
       */
      private String operationMethod;

      /**
       * 请求URL
       */
      private String requestUrl;

      /**
       * 请求方法
       */
      private String requestMethod;

      /**
       * 请求参数
       */
      private String requestParams;

      /**
       * 响应结果
       */
      private String responseResult;

      /**
       * 操作IP
       */
      private String ipAddress;

      /**
       * 操作地点
       */
      private String location;

      /**
       * 操作结果 1成功 0失败
       */
      private Integer operationResult;

      /**
       * 操作结果消息
       */
      private String resultMessage;

      /**
       * 閿欒鐮?
       */
      private String errorCode;

      /**
       * 寮傚父绫诲瀷
       */
      private String exceptionClass;

      /**
       * 操作时间
       */
      private Date operationTime;

      /**
       * 创建时间
       */
      private Date createTime;

      /**
       * 删除标记
       */
      @TableLogic
      private Integer deleted;
}
